plugin/types.go

```diff
@@ -58,6 +59,16 @@ type ClientHook interface {
 	Resource() client.Object
 }
 
+type Interceptor interface {
+	syncertypes.Base
+
+	// Handler is the handler that will handle the requests delegated by the syncer
+	Handler() http.Handler
+
+	// Config returns an rbac style struct which defines what to intercept
+	Config() v2.Interceptor
```

Comment:
I would call this `InterceptRequests()` and this should return an array of interceptors